lib/versioning/composer/index.spec.ts

```diff
@@ -148,6 +156,36 @@ describe('semver.getNewValue()', () => {
       })
     ).toEqual('^1.0');
   });
+  it('bumps caret to same', () => {
+    expect(
+      semver.getNewValue({
+        currentValue: '^1.0.0',
+        rangeStrategy: 'bump',
+        fromVersion: '1.0.0',
+        toVersion: '1.3.5',
+      })
+    ).toEqual('^1.0');
+  });
+  it('replaces caret to same', () => {
+    expect(
+      semver.getNewValue({
+        currentValue: '^1.0.0',
+        rangeStrategy: 'replace',
+        fromVersion: '1.0.0',
+        toVersion: '1.3.5',
+      })
+    ).toEqual('^1.0');
```

Comment:
This is not correct. I don't think you will get this query within Renovate though, because if the existing value is `^1.0.0` then `1.3.5` won't trigger an update. Ideally it would return `^1.0.0` if so

Reply:
Okay, so `getNewValue` won't be called at all if the update is satisfied in existing constraint? I copied this from this test:
```
    expect(
      semver.getNewValue({
        currentValue: '~4',
        rangeStrategy: 'replace',
        fromVersion: '4.0.0',
        toVersion: '4.2.0',
      })
    ).toEqual('~4');
```

Reply:
I think this might be a subtle behaviour based on if the constraint covers the entire version range as if it were `~`, not `^`. I'll try that out.

Reply:
In that case, let’s make sure it’s right and returns the existing value if it satisfies.